hibernate-core/src/main/java/org/hibernate/dialect/OracleServerConfiguration.java

```diff
@@ -46,39 +57,86 @@ public int getDriverMinorVersion() {
 	}
 
 	public OracleServerConfiguration(boolean autonomous, boolean extended) {
-		this( autonomous, extended, 19, 0 );
+		this( autonomous, extended, false, 19, 0 );
 	}
 
 	public OracleServerConfiguration(
 			boolean autonomous,
 			boolean extended,
 			int driverMajorVersion,
 			int driverMinorVersion) {
+		this(autonomous, extended, false, driverMajorVersion, driverMinorVersion);
+	}
+		
+	public OracleServerConfiguration(
+			boolean autonomous,
+			boolean extended,
+			boolean applicationContinuity,
+			int driverMajorVersion,
+			int driverMinorVersion) {
 		this.autonomous = autonomous;
 		this.extended = extended;
+		this.applicationContinuity = applicationContinuity;
 		this.driverMajorVersion = driverMajorVersion;
 		this.driverMinorVersion = driverMinorVersion;
 	}
 
 	public static OracleServerConfiguration fromDialectResolutionInfo(DialectResolutionInfo info) {
 		Boolean extended = null;
 		Boolean autonomous = null;
+		Boolean applicationContinuity = null;
 		Integer majorVersion = null;
 		Integer minorVersion = null;
 		final DatabaseMetaData databaseMetaData = info.getDatabaseMetadata();
 		if ( databaseMetaData != null ) {
 			majorVersion = databaseMetaData.getDriverMajorVersion();
 			minorVersion = databaseMetaData.getDriverMinorVersion();
 
-			try (final Statement statement = databaseMetaData.getConnection().createStatement()) {
-				final ResultSet rs = statement.executeQuery(
-						"select cast('string' as varchar2(32000)), " +
-								"sys_context('USERENV','CLOUD_SERVICE') from dual"
-				);
-				if ( rs.next() ) {
-					// succeeded, so MAX_STRING_SIZE == EXTENDED
-					extended = true;
-					autonomous = isAutonomous( rs.getString( 2 ) );
+
+			try {
+				final Connection c = databaseMetaData.getConnection();
+
+				// Use Oracle JDBC replay statistics information to determine if this
+				// connection is protected by Application Continuity
+				try {
+					final ReplayableConnection re = (ReplayableConnection) c;
```

Comment:
The problem with this way of detection is that you refer to classes which might not be visible to Hibernate. Is there no easier way to determine if this feature is enabled? Also, I would prefer if we reduce the amount of statements to execute if possible, which should be easily doable in this case i.e. you can use the statement we had before instead of executing a new dummy statement.

Reply:
Modified accordingly, thanks for the review!